lib/gen_stage/metrics.ex

```diff
@@ -0,0 +1,15 @@
+defmodule GenStage.Metrics do
+  @moduledoc false
+
+  use Instruments.CustomFunctions, prefix: "genstage"
+
+  @doc """
+  Returns the registered name of the current process or nil.
+  """
+  def metrics_self_name() do
+    case :erlang.process_info(self(), :registered_name) do
```

Comment:
prefer `Process.info/2` vs `:erlang.process_info/2` they are the same thing, but the former is the more "elixir" way to do it!

Reply:
right, didn't realize they were identical thanks! done